ui/src/containers/AppDetailsContainer.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import AppsTable from '../components/AppsTable';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { Checkbox } from '@patternfly/react-core';
+import { getApps, reverseAppDetailsTableSort } from '../actions/actions-ui';
+
+export class AppDetailsContainer extends React.Component {
+  componentWillMount () {}
+
+  handleChange = (e) => {
+    console.log('checkbox clicked');
+  };
+
+  onSort = (_event, index) => {
+    this.props.reverseAppDetailsTableSort(index);
+  };
+
+  getTable = () => {
+    const renderedRows = [];
+    for (let i = 0; i < this.props.apps.length; i++) {
+      const tempRow = [];
+      tempRow[0] = this.props.apps[i][0];
+      tempRow[1] = this.props.apps[i][1];
+      tempRow[2] = this.props.apps[i][2];
+      tempRow[3] = this.props.apps[i][3];
+      const cb = (
+        <React.Fragment>
+          <Checkbox
+            label=""
+            isChecked={this.props.apps[i][4]}
+            onChange={this.handleChange}
+            aria-label="controlled checkbox example"
+            id={i.toString()}
+          />
+        </React.Fragment>
+      );
+      tempRow[4] = cb;
+      tempRow[5] = this.props.apps[i][5];
+      renderedRows.push(tempRow);
+    }
+
+    return (
+      <div className="apps-table">
+        <AppsTable
+          caption="Deployed Versions"
+          columns={this.props.columns}
+          rows={renderedRows}
+          sortBy={this.props.sortBy}
+          onSort={this.onSort}
+          onRowClick={this.onRowClick}
+        />
+      </div>
+    );
+  };
+
+  render () {
+    if (this.props.isAppsRequestFailed) {
+      return (
+        <div className="no-apps">
+          <p>Unable to fetch any apps</p>
+        </div>
+      );
+    }
+    return this.getTable();
+  }
+}
+
+AppDetailsContainer.propTypes = {
+  apps: PropTypes.array.isRequired,
+  sortBy: PropTypes.object.isRequired,
+  columns: PropTypes.array.isRequired,
+  isAppsRequestFailed: PropTypes.bool.isRequired
+};
+
+function mapStateToProps (state) {
+  return {
+    apps: state.appDetailRows,
```

Comment:
should this be `app` rather than `apps`?

Reply:
It can be, but keeping it as `apps` reduces some of the differences in the containers, in case we choose to abstract the container component in future. For that reason I'd keep it `apps` for the moment, as each app version is technically an app anyway so don't think its too confusing

Reply:
Based on your comment above, if we are not going to abstract it for now, why not let's go with app. I think it's more accurate as to what's in the variable.

Reply:
@laurafitzgerald I don't think `app` would be accurate - as this is a list of app versions (with their associated details). We could rename it to `appDetails` but don't think we really gain anything. Let's keep the containers similar enough for now, as if we need to abstract out a common component in the future it will be easier.